File: backend/src/main/java/com/manageit/manageit/dto/project/ProjectDto.java
package com.manageit.manageit.dto.project;

import com.manageit.manageit.dto.user.BasicUserDto;
import com.manageit.manageit.project.ProjectStatus;
import com.manageit.manageit.dto.task.TaskDto;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProjectDto {
    private UUID id;
    private BasicUserDto owner;
    private String name;
    private String description;
    private ProjectStatus status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private LocalDate startDate;
    private LocalDate endDate;
    private Integer completedTasks;
    private Integer totalTasks;
    private List<TaskDto> tasks;
    private List<BasicUserDto> members;
}

File: frontend/src/app/features/dto/project.dto.ts
import { TaskDto } from '@/app/features/dto/task.dto';
import { ProjectStatus } from '@/app/modules/projects/types/project-status.type';
import { UserSummaryDto } from '@/app/shared/dto/user-summary.dto';

export interface ProjectDto {
  readonly id: string;
  readonly name: string;
  readonly description: string;
  readonly createdAt: string;
  readonly endDate: string;
  readonly completedTasks: number;
  readonly totalTasks: number;
  readonly status: ProjectStatus;
  readonly owner: UserSummaryDto;
  readonly members: UserSummaryDto[];
  readonly tasks: TaskDto[];
}

export interface ProjectPayload {
  readonly name: string;
  readonly description: string;
  readonly endDate: string;
}
